client/keystore/src/lib.rs

```diff
@@ -276,8 +298,65 @@ impl Store {
 }
 
 impl BareCryptoStore for Store {
+	fn supported_keys(
+		&self,
+		id: KeyTypeId,
+		keys: Vec<CryptoTypePublicPair>
+	) -> std::result::Result<HashSet<CryptoTypePublicPair>, TraitError> {
+		let keys = keys.into_iter().collect::<HashSet<_>>();
+		let all_keys = self.keys(id)?;
```

Comment:
I think it would be much faster if you directly check the filesystem etc for the incoming keys, instead of loading *all* keys and doing an intersection.

Reply:
Well, that's sort of the opposite what I suggested :) Aren't we going to have like thousands keys in `Vec` and then only couple in the storage? Also this can be optimized to not read from disk if needed.